drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/reason_for_exit_loader.rb

```diff
@@ -0,0 +1,90 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# matriculation to new platform
+module HmisExternalApis::AcHmis::Importers::Loaders
+  class ReasonForExitLoader < CustomDataElementBaseLoader
+    def filename
+      # note filename is slightly different from the spec
+      'ReasonforExit.csv'
+    end
+
+    protected
+
+    def cde_definitions_keys
+      [
+        :reason_for_exit_type,
+        :reason_for_exit_other,
+        :reason_for_exit_voluntary,
+        :reason_for_exit_involuntary,
+      ]
+    end
+
+    def build_records
+      exit_lookup = owner_class
+        .where(data_source: data_source)
+        .pluck(:exit_id, :id, :enrollment_id)
+        .to_h { |exit_id, pk, enrollment_id| [exit_id, [pk, enrollment_id]] }
+      rows.flat_map do |row|
+        exit_id = row_value(row, field: 'ExitID')
+        owner_id, enrollment_id = exit_lookup[exit_id]
+
+        unless exit_id
+          log_skipped_row(row, field: 'ExitID')
+          next # early return
+        end
+
+        raise 'ExitID/EnrollmentID mismatch' if enrollment_id != row_value(row, field: 'EnrollmentID')
```

Comment:
```suggestion
        unless owner_id
          log_skipped_row(row, field: 'ExitID')
          next # early return
        end

        raise 'ExitID/EnrollmentID mismatch' if enrollment_id != row_value(row, field: 'EnrollmentID')
```

Right? I'm getting an `ExitID/EnrollmentID mismatch` error, but I think its really just that the exit id wasnt found.

Reply:
Yes, that's right. Sorry about that!